kv: update uses of btree

RaduBerinde opened the pull request:
Use the newer btree implementation with generics.

Informs: #144504
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/144604)
<!-- Reviewable:end -->

tbg commented on pkg/kv/kvserver/scanner_test.go:
```diff
@@ -87,14 +90,14 @@ func (rs *testRangeSet) EstimatedCount() int {
 	rs.Lock()
 	defer rs.Unlock()
-	repl := rs.replicasByKey.Delete((rangeBTreeKey)(endKey))
+	repl, _ := rs.replicasByKey.Delete(&Replica{startKey: endKey})
```
why not keep using rangeBTreeKey here? Seems cleaner.

RaduBerinde replied:
Switched to btreemap.
<!-- Reviewable comment -OOwpC7q0vdC69T4Octj:b-gdqrcb -->
<!-- Sent from Reviewable.io -->

tbg commented on pkg/kv/kvserver/store_replica_btree.go:
```diff
@@ -268,7 +275,7 @@ type rangeKeyItem interface {
 
 // rangeBTreeKey is a type alias of roachpb.RKey that implements the
-// rangeKeyItem interface and the btree.Item interface.
+// rangeKeyItem interface.
```
Since you're here, could add here that its purpose is to look up contents of the map only (i.e. use for Ascend, Delete, etc, but never to be inserted).

RaduBerinde replied:
Switched to using `btreemap` which makes all this unnecessary (this is a fork of btree I made that separates the key and value, and we're already using it in other places).
<!-- Reviewable comment -OOwpE-M2tT0qcfyDWmK:bxjodid -->
<!-- Sent from Reviewable.io -->

tbg reviewed (approved):
LGTM, thank you!

RaduBerinde committed:
kv: update uses of btree

Use the newer btree implementation with generics.

Informs: #144504
Release note: None

RaduBerinde committed:
kvserver: switch to btreemap for store replica map

Epic: none
Release note: None

RaduBerinde reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/144604)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @tbg)



<!-- Sent from Reviewable.io -->

RaduBerinde commented:
bors r+
